dandiapi/api/views/dandiset.py

```diff
@@ -502,3 +512,59 @@ def clear_uploads(self, request, dandiset__pk):
 
         dandiset.uploads.all().delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+    @swagger_auto_schema(
+        methods=['POST'],
+        manual_parameters=[DANDISET_PK_PARAM],
+        request_body=no_body,
+        responses={
+            200: 'Dandiset starred successfully',
+            401: 'Authentication required',
+        },
+        operation_summary='Star a dandiset.',
+        operation_description='Star a dandiset. User must be authenticated.',
+    )
+    @action(methods=['POST'], detail=True)
+    def star(self, request, dandiset__pk) -> Response:
```

Comment:
I think this PR should be merged after #2115, and one of the reasons is that this PR could take advantage of the new permission service layer. For example, a decorator that requires authentication could be used, similar to the `require_dandiset_owner_or_403` decorator introduced in that PR.